packages/loader/container-loader/src/test/connectionManager.spec.ts

```diff
@@ -17,96 +17,122 @@ import { IConnectionManagerFactoryArgs } from "../contracts";
         assert.equal(connectionCount, 2, "Expected 2 connections after the first emitted an error");
 
         // Act II - nonretryable disconnect
         const disconnectReason: IAnyDriverError =
             new NonRetryableError("fatalDisconnectReason", undefined, DriverErrorType.genericError, { driverVersion: pkgVersion });
-        oldDeltaConnection = mockDeltaConnection;
-        mockDeltaConnection.emitDisconnect(disconnectReason);
-        await waitForConnection(3);
+        oldConnection = connection;
+        connection.emitDisconnect(disconnectReason);
+        connection = await waitForConnection();
 
         // Assert II
-        assert(oldDeltaConnection.disposed, "Old connection should be disposed after emitting disconnect");
-        assert.equal(mockDeltaConnection.clientId, `mock_client_${nextClientId - 1}`, "New connection should have expected id");
-        mockLogger.assertMatchAny([{ eventName: "reconnectingDespiteFatalError", reconnectMode: "Enabled", error: "fatalDisconnectReason", canRetry: false }]);
+        assert(oldConnection.disposed, "Old connection should be disposed after emitting disconnect");
+        assert.equal(connection.clientId, "mock_client_2", "New connection should have expected id");
+        mockLogger.assertMatchAny([{ eventName: "reconnectingDespiteFatalError", reconnectMode: "Enabled", error: "fatalDisconnectReason", canRetry: false, }]);
         assert(!closed, "Don't expect closeHandler to be called even when connection emits a non-retryable disconnect");
         assert.equal(disconnectCount, 2, "Expected 2 disconnects from emitting an error and disconnect");
         assert.equal(connectionCount, 3, "Expected 3 connections after the two disconnects");
 
         // Act III - nonretryable nack
         const nack: Partial<INack> = { content: { code: 403, type: NackErrorType.BadRequestError, message: "fatalNack"} };
-        oldDeltaConnection = mockDeltaConnection;
-        mockDeltaConnection.emitNack("docId", [nack]);
+        oldConnection = connection;
+        connection.emitNack("docId", [nack]);
 
         // Assert III
-        assert(oldDeltaConnection.disposed, "Old connection should be disposed after emitting nack");
-        assert.equal(mockDeltaConnection.clientId, `mock_client_${nextClientId - 1}`, "New connection should have expected id");
-        mockLogger.assertMatchAny([{ eventName: "reconnectingDespiteFatalError", reconnectMode: "Enabled", statusCode: 403, canRetry: false }]);
+        assert(!oldConnection.disposed, "connection shouldn't be disposed since mock closeHandler doesn't do it - don't expect it here after fatal nack");
+        assert.equal(connection, oldConnection, "Should not have gotten a new connection after fatal nack");
+        mockLogger.assertMatch([], "Expected no logs sent, specifically not reconnectingDespiteFatalError event");
         assert(closed, "closeHandler should be called in response to 403 nack");
```

Comment:
```suggestion
        assert(closed, "closeHandler should be called in response to 403 nack");
        assert(!oldConnection.disposed, "connection shouldn't be disposed since mock closeHandler doesn't do it - don't expect it here after fatal nack");
        mockLogger.assertMatch([], "Expected no logs sent, specifically not reconnectingDespiteFatalError event");
```